uiowa/uiowa
# P2LB  - Prevent access to paragraphs widget in new nodes

Relates to https://github.com/uiowa/uiowa/issues/7049

# Problem

New page content that is created with the p2lb converter on thinks it is a v2 page the next time it is edited because a v3 revision id is never set for the node.

This PR does not address nodes that got created since P2LB got turned on. Instead it is focused on limiting the scope of the problem to prevent new instances where the problem described above can happen.

# To test
- Code review
```
ddev blt ds --site law.uiowa.edu && ddev drush @law.local uli node/add/page
```
- Add a page and name it something like "After P2LB" so you can easily identify it later.
- Click the "Edit" tab and confirm that you are not able to see the v2 Section Blocks field.
- Click the "V3 Status" tab and confirm that the message you see says "This is a v3 page that was created after the SiteNow v2 to v3 converter was enabled."
- Confirm that you can see/click the "Layout" tab and add blocks through LB.
- Visit https://law.uiowa.ddev.site/admin/content/sitenow-converter and confirm that the new page does not show up in the list.
- Confirm the number of results matches https://law.uiowa.edu/admin/content/sitenow-converter
- Visit https://law.uiowa.ddev.site/admin/content/sitenow-converter/converted and confirm that the new page does not show up in the list.
- Confirm the number of results matches https://law.uiowa.edu/admin/content/sitenow-converter/converted. The quick way of doing this is to find the last page of results, make sure it matches and then count just the results on the last page.
- Presumably, you've created some revisions by this point to the page you created. If not, edit the page to add a revision.
- Confirm that you can delete all previous revisions.
- Confirm that you can delete the node you created.

## Cleanup
**Note: Running the cleanup script is currently not working on HMB for reasons that I'm not entirely clear about. I think we could potentially cover the code change related to that with a code review since it seems like we'll be investing some more time in fixing it regardless.**
```
ddev blt ds --site hawkeyemarchingband.uiowa.edu && ddev drush @hawkeyemarchingband.local config-split:activate p2lb && ddev drush @hawkeyemarchingband.local cim -y && ddev drush @hawkeyemarchingband.local uli node/add/page
```
- Add a new page. Edit it a few times to create some revisions.
- https://hawkeyemarchingband.uiowa.ddev.site/admin/content/sitenow-converter
- Convert all nodes
- Publish all converted nodes
```
ddev drush @hawkeyemarchingband.local sitenow_p2lb:cleanup
```
- Confirm that the page you created has not had its revisions removed.

<details>
<summary>Original problem/testing instructions</summary>

# Problem

New page content that is created with the p2lb converter on thinks it is a v2 page the next time it is edited because a v3 revision id is never set for the node.

By setting a numeric value for this field for new content, the existing logic works. The v3 page will show in the converted section but reverting it will have no effect. This field is removed when the converter is removed so no bad data will persist after conversion.

For existing nodes, It looks like nodes that are _not_ empty when it comes to layout_builder__layout but _are_ empty when it comes to field_v3_conversion_revision_id. Setting it to the current revision id allows p2lb:clean up to successfully delete any v2 paragraphs that got caught in the mix.

# To Test

On `main` sync law `ddev blt ds --site=law.uiowa.edu` and create a new page. You see the body field when saving the first time, but then go and edit it after saving. Because the field was never set, you get the v2 paragraphs fields and it thinks the page has not yet been converted.

Create a temp view that has the following filters:

Content: Content type (= Page)
Content: Conversion revision ID (empty)
Content: Layout (not empty)

See the results if any. If Law, there should be a couple.

On `hotfix_p2lb_new_page_default` run `ddev drush @law.local updb && ddev drush @law.local cim`

First check content overview to see if any page nodes were updated. They should have "automated save" revision entry by user 1. The temp view, if it had any results, should now be empty.
Second create a new page again. Editing this new page should keep the body field and not show you the v2 paragraphs fields. Find the page in the converted section and try to revert back to v2. It silently fails without error to the user.

</details>
